api/r/cellxgene.census/R/open.R

```diff
@@ -1,47 +1,63 @@
-DEFAULT_TILEDB_CONFIGURATION <- c(
-  "py.init_buffer_bytes" = paste(1 * 1024**3),
-  "soma.init_buffer_bytes" = paste(1 * 1024**3)
-)
-
 #' Open the Cell Census
 #'
 #' @param census_version The version of the Census, e.g., "latest".
 #' @param uri The URI containing the Census SOMA objects. If specified, takes
 #'            precedence over `census_version`.
 #' @param tiledbsoma_ctx A custom `tiledbsoma::SOMATileDBContext`
 #'
-#' @return Top-level `tiledbsoma::SOMACollection` object
+#' @return Top-level `tiledbsoma::SOMACollection` object. After use, the census
+#'         should be closed to release memory and other resources, usually with
+#'         `on.exit(census$close(), add = TRUE)`. Closing the top-level census
+#'         will also close all SOMA objects accessed through it.
 #' @importFrom tiledbsoma SOMACollection
 #' @importFrom tiledbsoma SOMATileDBContext
 #' @export
 #'
 #' @examples
 open_soma <- function(census_version = "latest", uri = NULL, tiledbsoma_ctx = NULL) {
-  s3_region <- NULL
-
-  if (is.null(uri)) {
+  if (is.null(uri) || is.null(tiledbsoma_ctx)) {
     description <- get_census_version_description(census_version)
-    uri <- description$soma.uri
-    if ("soma.s3_region" %in% names(description) &&
-      description$soma.s3_region != "") {
-      s3_region <- description$soma.s3_region
+    if (is.null(uri)) {
+      uri <- description$soma.uri
+    }
+    if (is.null(tiledbsoma_ctx)) {
+      tiledbsoma_ctx <- new_SOMATileDBContext_for_census(description)
     }
   }
 
+  return(tiledbsoma::SOMACollectionOpen(uri, tiledbsoma_ctx = tiledbsoma_ctx))
+}
+
+DEFAULT_TILEDB_CONFIGURATION <- c(
+  "py.init_buffer_bytes" = paste(1 * 1024**3),
+  "soma.init_buffer_bytes" = paste(1 * 1024**3)
+)
+
+new_SOMATileDBContext_for_census <- function(census_version_description, ...) {
```

Comment:
this is missing `vfs.s3.no_sign_request` which is needed to enable anonymous access

Reply:
@ebezzi has #471 open on this...@ebezzi how is that doing or can I help get it ready?

Reply:
The PR was working but I hit an issue in the `download_source_h5ad` test since the AWS library we use doesn't support anonymous access (not as far as I could find - see [this Slack thread](https://czi-sci.slack.com/archives/C048BN7K6U8/p1684266483832919)). Let's get this merged so that the CI is green again and then I can figure out how to fix the test in my PR with a workaround.